src/main/java/com/conveyal/taui/persistence/LocalFilePersistence.java

```diff
@@ -0,0 +1,63 @@
+package com.conveyal.taui.persistence;
+
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.conveyal.taui.util.JsonUtil;
+import spark.Response;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+
+public class LocalFilePersistence extends FilePersistence {
+    private final String baseDirectory;
+
+    private String format (String directory, String path) {
+        return String.format("%s/%s/%s", baseDirectory, directory, path);
+    }
+
+    public LocalFilePersistence(String directory) {
+        baseDirectory = directory;
+        File bd = new File(baseDirectory);
+        bd.mkdir();
+    }
+
+    public File create (String directory, String path) {
+        File file = new File(format(directory, path));
+        // Ensure the directories have been made
+        file.mkdirs();
+
+        return file;
+    }
+
+    public boolean exists (String directory, String path) {
+        return new File(format(directory, path)).exists();
+    }
+
+    public Object put(String directory, String path, Object data) throws IOException {
```

Comment:
This method returns a File but the S3-backed implementation returns a PutObjectResult. The interface should be more strictly typed.